airflow/models.py

```diff
@@ -3055,6 +3055,34 @@ def latest_execution_date(self):
         session.close()
         return execution_date
 
+    def descendants(self, dagbag, task_ids=None, include_downstream=False,
```

Comment:
It probably is smart to be defensive against cyclic dependencies.

Reply:
@bolkedebruin Do you think adding this check when the dagbag is being populated is better? That way the invariant of the dagbag being well-formed is maintained, both at the dag level, and at the higher level.

Of course, since the process that refreshes the dags is called way more often than this `descendants()` method, overall there might be a slight perf hit in that approach. Maintaining a "valid" dagbag always however sounds more robust to me.

What do you think?